src/Std/Tactic/BVDecide/Bitblast/BVExpr/Circuit/Lemmas/Expr.lean

```diff
@@ -229,7 +229,9 @@ theorem go_denote_eq (aig : AIG BVBit) (expr : BVExpr w) (assign : Assignment) :
   | un op expr ih =>
     cases op with
     | not => simp [go, ih, hidx]
-    | shiftLeftConst => simp [go, ih, hidx]
+    | shiftLeftConst i => simp only [go, denote_blastShiftLeftConst, ih, dite_eq_ite,
+      Bool.if_false_left, eval_un, BVUnOp.eval_shiftLeftConst, hidx, BitVec.getLsbD_eq_getElem,
+      BitVec.getElem_shiftLeft]; congr;
```

Comment:
How can I name `w` in this proof state? It somehow gets lost. If I have it, the proof can be simpler.

Reply:
If the type of the desired `w` is enough to specify it, then `rename` is probably the cleanest option.

Reply:
I found uses of `rename_i` in the same function, so it seems this is permitted. Thank you for the pointer. I now used it to simplify the code.